app/src/main/java/com/navapp/data/DefaultStoreDao.java

```diff
@@ -17,8 +19,18 @@
 
     @Insert
     void insert(DefaultStoreModel store);
+    @Insert(onConflict = OnConflictStrategy.IGNORE)
+    long tryInsert(DefaultStoreModel store);
     @Update
     void update(DefaultStoreModel store);
     @Delete
     void delete(DefaultStoreModel store);
+
+    @Transaction
+    default void upsert(DefaultStoreModel model) {
+        long id = tryInsert(model);
+        if (id == -1) {
+            update(model);
+        }
+    }
```

Comment:
It tries to insert and if id == -1 it updates, what does that mean? also, if it's conditional, weird to call it this way. (besides the cool word)

Reply:
https://sqlite.org/lang_UPSERT.html It's the actual name....

The `defaults` is structured such that each row has defined by a key which marks which table the default is for. 

When we want to set a default to the table, we would have to `insert` if the row doesn't exist, or `update` if it does. Trying to `insert` a row whose key already exists will result in a `conflict` and fail. In such a case, `-1` is returned, and we need to use `update` instead.